### PR TITLE
feat(tools): add generate fragment types builder

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Daffodil does not provide a packaged way to generate GraphQL fragment data.

## What is the new behavior?
Daffodil provides a new package `@daffodil/builders` which contains a builder for generating GraphQL fragment type data.

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information
Tests are causing issues at the moment so they have been omitted.

### CODE REVIEW

**Comment by lderrickable** [comment]
What do you mean by this? `Tests are causing issues at the moment so they have been omitted.`

**Comment by griest024** [comment]
@lderrickable Importing the `@graphql-codegen/cli` package in the testing environment is what the root of the issues are, I think. It balloons the compilation to 2500+ modules and throws a million lines of errors to STDOUT, mostly about browser not being a valid alias configuration or something like that.

**Comment by lderrickable** [comment]
Could you make it so that someone can follow the docs and it works? I followed the docs and it doesn't seem to work, but maybe I'm doing something wrong. I don't see how it could work though because the build script references `@daffodil/builders` but I don't think that's a thing

**Comment by griest024** [comment]
I don't think I have perms to publish `@daffodil` packages but this would be a published package. For testing I used `npm link` to test. I can open a PR with an implementation in demo if it makes it easier.

**Comment by lderrickable** [comment]
We wouldn't publish it until after it got pushed to develop anyway. If it is publishable, we should be able to still run it from the dist folder, right? There is currently no way to build it into the dist folder I don't think.

**Comment by griest024** [comment]
Its dependent on file structure so I left it building to its own personal dist folder for initial testing. I'm not sure how the publish/release process works with daffodil packages but I can make it build to the main dist folder if that is a necessary part of it. I think it needs to read `package.json` in order to run the builder though, so just running based off what is in the `dist` folder is not possible.

File: package.json [hunk 1]
```diff
@@ -52,13 +52,16 @@
     "zone.js": "~0.9.1"
   },
   "devDependencies": {
+    "@angular-devkit/architect": "^0.901.0",
```
Comment (damienwebdev) [inline package.json:55]: Can you pull this down to pre-ivy?

Comment (griest024) [reply package.json]: changed

**Pull request merged**